java/sage/api/CaptureDeviceAPI.java

```diff
@@ -311,12 +311,50 @@ public Object runSafely(Catbert.FastStack stack) throws Exception{
           ((DShowCaptureDevice) cdev).setDTVStandard(std);
         }
         return null;
+      }});
+    rft.put(new PredefinedJEPFunction("CaptureDevice", "GetCaptureDeviceMerit", new String[] {"CaptureDevice"}, true)
+    {
+      /**
+       * Gets the encoder merit for a CaptureDevice.
+       * @param CaptureDevice the name of the CaptureDevice
+       * @return The merit value for the specified CaptureDevice; If all else is equal, a capture device with a higher merit has higher priority.
+       *
+       * @declaration public int GetCaptureDeviceMerit(String CaptureDevice);
+       */
+      public Object runSafely(Catbert.FastStack stack) throws Exception{
+        String capDevName = getString(stack); 
+        CaptureDevice capDev = getCapDevObj(capDevName);
+        if (capDev == null)
+        {
+          CaptureDeviceInput capDevInput = getCapDevInputObj(capDevName);

```

Comment:
If you handled this case in getCapDevObj then you wouldn't need the code here or in your comparator. :)